arangod/Replication2/Methods.cpp

```diff
@@ -320,7 +320,48 @@ struct ReplicatedLogMethodsCoordinator final
 
   auto insert(LogId id, TypedLogIterator<LogPayload>& iter) const
       -> futures::Future<std::pair<std::vector<LogIndex>, replicated_log::WaitForResult>> override {
-    THROW_ARANGO_EXCEPTION(TRI_ERROR_NOT_IMPLEMENTED);
+    auto path = basics::StringUtils::joinT("/", "_api/log", id, "multi-insert");
+
+    std::size_t payloadSize{0};
+    VPackBuilder builder{};
+    {
+      VPackArrayBuilder arrayBuilder{&builder};
+      while (const auto payload = iter.next()) {
+        if (payload.has_value()) {
+          builder.add(payload->slice());
+        } else {
+          builder.add(VPackSlice::emptyObjectSlice());
+        }
+        ++payloadSize;
+      }
+    }
+
+    network::RequestOptions opts;
+    opts.database = vocbase.name();
+    return network::sendRequest(pool, "server:" + getLogLeader(id),
+                                fuerte::RestVerb::Post, path, builder.bufferRef(), opts)
+        .thenValue([payloadSize](network::Response&& resp) {
+          if (resp.fail() || !fuerte::statusIsSuccess(resp.statusCode())) {
+            THROW_ARANGO_EXCEPTION(resp.combinedResult());
+          }
+          auto result = resp.slice().get("result");
+          auto waitResult = result.get("result");
+
+          auto quorum = std::make_shared<replication2::replicated_log::QuorumData const>(
+              waitResult.get("quorum"));
+          auto commitIndex = waitResult.get("commitIndex").extract<LogIndex>();
+
+          std::vector<LogIndex> indexes;
+          indexes.reserve(payloadSize);
+          auto indexIter = velocypack::ArrayIterator(result.get("indexes"));
```

Comment:
I just noticed that the http-api returns something different than the js api does (this applies to the single insert function as well). In JS on the server (i.e. using V8) we have only the `index` or `indexes` field, in Http and therefore client-js there are information about the quorum. We should include the quorum information in V8 as well.